src/schema/artist/insights.ts

```diff
@@ -0,0 +1,88 @@
+import { compact } from "lodash"
+
+import {
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLEnumType,
+  GraphQLString,
+} from "graphql"
+
+export const InsightType = new GraphQLEnumType({
+  name: "InsightType",
+  values: {
+    SOLO_SHOW: {
+      value: "Solo show at a major institution",
+    },
+    GROUP_SHOW: {
+      value: "Group show at a major institution",
+    },
+    COLLECTED: {
+      value: "Collected by a major institution",
+    },
+    REVIEWED: {
+      value: "Reviewed by a major art publication",
+    },
+    BIENNIAL: {
+      value: "Participated in a major biennial",
+    },
+  },
+})
+
+const ArtistInsightTypeMapping = {
+  major_solo: InsightType.getValue("SOLO_SHOW"),
+  major_group: InsightType.getValue("GROUP_SHOW"),
+  major_collected: InsightType.getValue("COLLECTED"),
+  major_review: InsightType.getValue("REVIEWED"),
+  major_biennial: InsightType.getValue("BIENNIAL"),
+}
+
+const buildInsights = artist => {
+  const splitEntities = (entities: String): Array<String> => {
+    return entities.split("|").map(entity => {
+      return entity.trim()
+    })
+  }
+
+  const buildInsight = (insightType, entitiesString: String) => {
+    return {
+      type: insightType.name,
+      entities: splitEntities(entitiesString),
+      label: insightType.value,
+    }
+  }
+
+  return compact(
+    Object.keys(ArtistInsightTypeMapping).map(key => {
+      if (artist[key]) {
+        return buildInsight(ArtistInsightTypeMapping[key], artist.major_solo)
+      }
+    })
+  )
+}
+
+const Insight = new GraphQLObjectType({
+  name: "Insight",
+  fields: {
+    type: {
+      type: GraphQLString,
+      description: "The type of insight.",
+    },
+    label: {
+      type: GraphQLString,
+      description: "Label to use when displaying the insight.",
+    },
+    entities: {
+      type: new GraphQLList(GraphQLString),
+      description: "List of entities relevant to the insight.",
+    },
+  },
+})
+
+const ArtistInsights = {
+  type: new GraphQLList(Insight),
+  resolve: artist => {
+    return buildInsights(artist)
+  },
+}
+
+export default ArtistInsights
```

Comment:
While this breaks protocol with much of how MP is written we've been trying to avoid default exports since they're not statically analyzable.

Reply:
Gotcha. I've switched to an explicitly named import :+1: